src/main/java/com/openlattice/util/SingletonArrays.java

```diff
@@ -0,0 +1,25 @@
+package com.openlattice.util;
+
+import java.util.UUID;
+
+public class SingletonArrays {
```

Comment:
Shouldn't this be EmptyArrays? Btw, why do we need this, isn't it easier to call <type>ArrayOf() in kotlin?

Reply:
`arrayOf()` doesn't return a reference to a singleton empty array, it just creates a new 0-sized array. 
`EmptyArrays` is probably a better name but I want it to be clear that they are singletons so `SingletonEmptyArrays` may be the best name

It's just a micro-optimization because 0-length arrays are basically placeholder type references b/c you can't resize arrays. So if you're calling `PostgresArrays.newIntArray( new Integer[0]);` a thousand times, that's 1000 object array allocations vs using this is 1000 pointer dereference operations which are a significantly faster operation

I was bored on Friday so I figured I'd throw this in here

Reply:
I see, thanks for the explanation!